dbos/_core.py

```diff
@@ -475,6 +479,15 @@ def start_workflow(
 
     new_wf_id, new_wf_ctx = _get_new_wf()
 
+    ctx = new_wf_ctx
+    new_child_workflow_id = ctx.id_assigned_for_next_workflow
+    if ctx and ctx.parent_workflow_id != "":
```

Comment:
I don't think the `if ctx` check is necessary, as `_get_new_wf` always returns a valid context.